torch_sim/runners.py

```diff
@@ -485,9 +501,13 @@ def optimize[T: OptimState](  # noqa: C901, PLR0915
             and og_filenames is not None
             and (step == 1 or len(converged_states) > 0)
         ):
+            mode_before = trajectory_reporter.trajectory_kwargs["mode"]
+            # temporarily set to "append" mode to avoid overwriting existing files
+            trajectory_reporter.trajectory_kwargs["mode"] = "a"
```

Comment:
Is the append mode fixing the bug or just necessary for appending?

More generally, this isn't grappling with the fact that some the existing trajectories might already exist or might have different numbers of steps